src/lib/util/value.c

```diff
@@ -1478,12 +1478,20 @@ ssize_t fr_value_box_from_network(TALLOC_CTX *ctx,
 				  uint8_t const *src, size_t len,
 				  bool tainted)
 {
-	uint8_t	const *p = src;
-	uint8_t const *end = p + len;
-	size_t	min, max;
+	return fr_value_box_from_network_dbuff(ctx, dst, type, enumv, &FR_DBUFF_TMP(src, len), tainted);
+}
+
+ssize_t fr_value_box_from_network_dbuff(TALLOC_CTX *ctx,
+					fr_value_box_t *dst, fr_type_t type, fr_dict_attr_t const *enumv,
+					fr_dbuff_t *dbuff,
```

Comment:
This should still take an explicit length parameter.  The length will usually come from the attribute's length field.  The reason why we'd want to it like this as it means we don't have to write boilerplate code for doing length checks.  We can pass the dbuff representing the rest of the data on the wire, and the length from the attribute header, and fr_value_box_from_network_dbuff should error if len exceeds the available data in the dbuff.

Reply:
Perhaps I don't understand. Isn't that length (more accurately the minimum and maximum length) retrieved from the attribute's type, which is already a parameter?

Reply:
I didn't understand. It should indeed stay.

Reply:
That change is now made and pushed.